mod/redir.php

```diff
@@ -70,7 +70,10 @@ function redir_init(App $a) {
 				&& is_array($_SESSION['remote']))
 			{
 				foreach ($_SESSION['remote'] as $v) {
-					if ($v['uid'] == $_SESSION['visitor_visiting'] && $v['cid'] == $_SESSION['visitor_id']) {
+					if (!empty($v['uid']) && !empty($_SESSION['visitor_visiting']) &&
```

Comment:
Please use `Session::get('visitor_visiting')` and `Session::get('visitor_id')` instead. It checks for session key existence and returns a default value (or `null` by default) if it doesn't.